check-disk/lib/check-disk.go

```diff
@@ -0,0 +1,148 @@
+package checkdisk
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/mackerelio/checkers"
+)
+
+type diskStatus struct {
+	All   uint64 `json:"all"`
+	Used  uint64 `json:"used"`
+	Free  uint64 `json:"free"`
+	Avail uint64 `json:"available"`
+}
+
+var opts struct {
+	Warning  *string `short:"w" long:"warning" value-name:"N, N%" description:"Exit with WARNING status if less than N units or N% of disk are free"`
+	Critical *string `short:"c" long:"critical" value-name:"N, N%" description:"Exit with CRITICAL status if less than N units or N% of disk are free"`
+	Path     *string `short:"p" long:"path" value-name:"PATH" description:"Mount point or block device as emitted by the mount(8) command"`
+	Units    *string `short:"u" long:"units" value-name:"STRING" description:"Choose bytes, kB, MB, GB, TB (default: MB)"`
+}
+
+const (
+	b  = float64(1)
+	kb = float64(1024) * b
+	mb = float64(1024) * kb
+	gb = float64(1024) * mb
+	tb = float64(1024) * gb
+)
+
+func getDiskUsage(path string) (*diskStatus, error) {
+	fs := syscall.Statfs_t{}
+	err := syscall.Statfs(path, &fs)
+	if err != nil {
+		return nil, err
+	}
+
+	disk := &diskStatus{}
+	disk.All = fs.Blocks * uint64(fs.Bsize)
+	disk.Free = fs.Bfree * uint64(fs.Bsize)
+	disk.Used = disk.All - disk.Free
+	disk.Avail = fs.Bavail * uint64(fs.Bsize)
+
+	return disk, nil
+}
+
+func checkStatus(val string, units float64, disk *diskStatus) (checkers.Status, error) {
+	avail := float64(disk.Avail) / float64(units)
+	freePct := (float64(disk.Avail) * float64(100)) / float64(disk.All)
+
+	checkSt := checkers.OK
+	if strings.HasSuffix(val, "%") {
+		v, err := strconv.Atoi(strings.TrimRight(val, "%"))
+		if err != nil {
+			return checkers.UNKNOWN, err
+		}
+
+		if float64(v) > freePct {
+			checkSt = checkers.WARNING
+		}
+	} else {
+		v, err := strconv.Atoi(val)
+		if err != nil {
+			return checkers.UNKNOWN, err
+		}
+
+		if float64(v) > avail {
+			checkSt = checkers.WARNING
```

Comment:
checkStatus is also used with `opts.Critical`, but this implementation never returns `checkers.CRITICAL`.
So you should do one of following:
- receive checker.Status param to use on exceeding threshold
- return some signature to tell exceeding, not checkers.Status itself